challenge_3/go/makernaren/max_occurance.go

```diff
@@ -0,0 +1,38 @@
+// Take an array of mixed data types and find the element that occurs
+// most number of times.
+package main
```

Comment:
I would put the challenge functionality and unit tests in a challenge specific package and break out main() into its own file / package.